policies.go

```diff
@@ -319,6 +319,8 @@ type HostSelectionPolicy interface {
 	// so it's safe to have internal state without additional synchronization as long as every call to Pick returns
 	// a different instance of NextHost.
 	Pick(ExecutableQuery) NextHost
+	// Checks if datacenter is valid if local aware policy is used (explicitly or as a fallback)
+	IsDatacenterValid(datacenters []string) bool
```

Comment:
But it actually just performs as single validation (today)? It'd be great if in the future we could think and maybe add additional validation, but I'm not aware of any?

Reply:
I thought the name describes exactly what the function checks, so at the moment it's good enough.

Reply:
What do you think of making it future proof and just feed `Session` to a policy to let it self check what it needs to check ?

Reply:
I'm against it, just because of time pressure, not because of correctness. I think the code right now is good enough, and we are under some pressure to deliver it.

Reply:
The change can be done at the same time.
(I also don't think the original issue put any time pressure here).

Reply:
This issue is not that urgent - this is a footgun, not something that needs immediate fix.
Changing the interface now is easy, changing it later is not because of backwards compatibility.

Reply:
Also, should we verify rack name too? Maybe a more general mechanism that can be extended later would be better.

Reply:
@mykaul , are you ok with changing api to `<>(s *Session) error` ?
And covering rack validation as well ?